client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/SutController.java

```diff
@@ -270,36 +296,155 @@ public final ExtraHeuristicsDto computeExtraHeuristics() {
         if (sqlHandler.isCalculateHeuristics() || sqlHandler.isExtractSqlExecution()){
             ExecutionDto executionDto = sqlHandler.getExecutionDto();
             dto.databaseExecutionDto = executionDto;
+            // set accessed table
+            if (executionDto != null){
+                accessedTables.addAll(executionDto.deletedData);
+                accessedTables.addAll(executionDto.insertedData.keySet());
+//                accessedTables.addAll(executionDto.queriedData.keySet());
+                accessedTables.addAll(executionDto.insertedData.keySet());
+                accessedTables.addAll(executionDto.updatedData.keySet());
+            }
         }
 
         return dto;
     }
 
+    private void getTableToClean(List<String> accessedTables, List<String> tablesToClean){
+        for (String t: accessedTables){
+            if (!findInCollectionIgnoreCase(t, tablesToClean).isPresent()){
+                if (findInMapIgnoreCase(t, fkMap).isPresent()){
+                    tablesToClean.add(t);
+                    List<String> fk = fkMap.entrySet().stream().filter(e->
+                            findInCollectionIgnoreCase(t, e.getValue()).isPresent()
+                                    && !findInCollectionIgnoreCase(e.getKey(), tablesToClean).isPresent()).map(Map.Entry::getKey).collect(Collectors.toList());
+                    if (!fk.isEmpty())
+                        getTableToClean(fk, tablesToClean);
+                }else {
+                    SimpleLogger.uniqueWarn("Cannot find the table "+t+" in ["+String.join(",", fkMap.keySet())+"]");
+                }
+
+            }
+        }
+    }
+
+
+    private Optional<String> findInCollectionIgnoreCase(String name, Collection<String> list){
+        return list.stream().filter(i-> i.equalsIgnoreCase(name)).findFirst();
+    }
+
+    private Optional<? extends Map.Entry<String, ?>> findInMapIgnoreCase(String name, Map<String, ?> list){
+        return list.entrySet().stream().filter(x-> x.getKey().equalsIgnoreCase(name)).findFirst();
+    }
+
+    public void cleanAccessedTables(){
```

Comment:
if public method, add JavaDoc to it. also, should be on top, and not after other private methods

Reply:
also, I guess should be marked with `final`